tw-tasks-core/src/main/java/com/transferwise/tasks/health/ClusterWideTasksStateMonitor.java

```diff
@@ -61,6 +63,9 @@
   private List<Object> registeredMetricHandles;
   private Map<String, Object> taskInErrorStateHandles;
 
+  private Lock stateLock = new ReentrantLock();
+  private volatile boolean initialized;
```

Comment:
Since read/write is guarded by stateLock we can make it non volatile

Reply:
Fixed.